pkg/thing/interactors/register_thing.go

```diff
@@ -1,45 +1,11 @@
 package interactors
 
 import (
+	"errors"
 	"fmt"
 	"strconv"
 )
 
-// ErrorIDLenght is raised when ID is more than 16 characters
-type ErrorIDLenght struct{}
-
-// ErrorIDInvalid is raised when ID is not in hexadecimal value
-type ErrorIDInvalid struct{}
-
-// ErrorNameNotFound is raised when Name is empty
-type ErrorNameNotFound struct{}
-
-// ErrorMissingArgument is raised when there is some argument missing
-type ErrorMissingArgument struct{}
-
-// ErrorInvalidTypeArgument is raised when the type is not the expected
-type ErrorInvalidTypeArgument struct{ msg string }
-
-func (err ErrorIDLenght) Error() string {
-	return "ID length error"
-}
-
-func (err ErrorIDInvalid) Error() string {
-	return "ID is not in hexadecimal"
-}
-
-func (err ErrorNameNotFound) Error() string {
-	return "Name not found"
-}
-
-func (err ErrorMissingArgument) Error() string {
-	return "Missing arguments"
-}
-
-func (err ErrorInvalidTypeArgument) Error() string {
-	return err.msg
-}
-
```

Comment:
Nice, also had a feeling about these errors :joy: 

I also saw [this](https://github.com/docker/distribution/blob/master/reference/reference.go#L40) approach and it seems reasonable:

1. Erros are grouped on the top of the file (better readability, lookup and scaling)
2. Erros are referenced in a more concise way (better readability and easier to manage when refactoring) 

What do you think about doing something like this? I guess its a middle ground between as it was before and this new approach. Also, the approach referenced seems to have a few advantages worth considering.

Reply:
@lcbm Agree with you! I think it's better because in test files we expect these errors to occur in use cases

Reply:
+1 on this approach, @lcbm.

Reply:
@lcbm @joaoaneto Done!